activejob/test/cases/argument_serialization_test.rb

```diff
@@ -188,6 +190,25 @@ def self.permitted?
     end
   end
 
+  test "should maintain a functional duration" do
+    duration = perform_round_trip([1.year]).first
+    assert_equal 2.years, duration + 1.year
+  end
+
+  # TODO: remove before 7.2
```

Comment:
Rather than a TODO which could be missed, could we just make this test fail if the Rails version is 7.2 or greater, with a message indicating to remove the functionality?

Reply:
I couldn't see any similar messages in the codebase, do we just want something like this?

```diff
  test "should be able to deserialize a duration serialized under the old scheme" do
+    if Rails.version >= "7.2"
+      raise "This backwards-compatibility method can be removed"
+    end

    old_serialization = [{
      "_aj_serialized" => "ActiveJob::Serializers::DurationSerializer",
      "value" => 31556952,
      "parts" => [[{
        "_aj_serialized" => "ActiveJob::Serializers::SymbolSerializer",
        "value" => "years"
      }, 1]]
    }]
    duration = ActiveJob::Arguments.deserialize(old_serialization).first
    assert_equal 2.years, duration + 1.year
  end
```